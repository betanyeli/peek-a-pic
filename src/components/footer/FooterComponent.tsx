import { Image, View } from 'react-native';
import React from 'react';
import styles from './FooterComponent.styles';

type FooterComponentProps = {
  children: JSX.Element;
  showLogo: boolean;
};
const FooterComponent = ({ children, showLogo }: FooterComponentProps) => {
  return (
    <View style={styles.container}>
      {children}
      {showLogo && (
        <Image
          style={styles.logo}
          source={require('../../assets/images/logo.png')}
        />
      )}
    </View>
  );
};

export default FooterComponent;
