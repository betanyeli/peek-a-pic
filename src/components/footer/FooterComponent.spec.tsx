import { render, waitFor } from '@testing-library/react-native';
import { View } from 'react-native';
import FooterComponent from './FooterComponent';
import React from 'react';

describe('FooterComponenttest', () => {
  it('Should be match with snapshot with logo', () => {
    //Arrange
    const children = <View />;
    const { toJSON } = render(<FooterComponent showLogo children={children} />);

    // Act
    waitFor(() => {
      expect(toJSON).toMatchSnapshot();
    });
  });

  it('Should be match with snapshot without logo', () => {
    //Arrange
    const children = <View />;
    const { toJSON } = render(
      <FooterComponent showLogo={false} children={children} />,
    );

    // Act
    waitFor(() => {
      expect(toJSON).toMatchSnapshot();
    });
  });
});
