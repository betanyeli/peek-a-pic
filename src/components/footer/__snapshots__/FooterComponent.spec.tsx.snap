// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FooterComponenttest Should be match with snapshot 1`] = `[Function]`;

exports[`FooterComponenttest Should be match with snapshot with logo 1`] = `[Function]`;

exports[`FooterComponenttest Should be match with snapshot without logo 1`] = `[Function]`;
